pkg/operator/staticpod/controller/common/helpers.go

```diff
@@ -0,0 +1,81 @@
+package common
+
+import (
+	"strings"
+
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"k8s.io/apimachinery/pkg/api/equality"
+	errutils "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/util/retry"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+)
+
+// UpdateStatus applies the update funcs to the oldStatus, tries to update via the client and repeats
+// up to 3 times to react to conflict errors.
+func UpdateStatus(client OperatorClient, resourceVersion string, oldStatus *operatorv1.StaticPodOperatorStatus, updateFuncs ...func(status *operatorv1.StaticPodOperatorStatus) error) (bool, error) {
+	updated := false
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		_, oldStatus, resourceVersion, err := client.Get()
+		if err != nil {
+			return err
+		}
+
+		newStatus := oldStatus.DeepCopy()
+		for _, update := range updateFuncs {
+			if err := update(newStatus); err != nil {
+				return err
+			}
+		}
+
+		if equality.Semantic.DeepEqual(oldStatus, newStatus) {
+			return nil
+		}
+
+		_, err = client.UpdateStatus(resourceVersion, newStatus)
+		updated = err == nil
+		return err
+	})
+
+	return updated, err
+}
+
+// SetCondition sets the condition in the operator status.
+func SetCondition(client OperatorClient, resourceVersion string, oldStatus *operatorv1.StaticPodOperatorStatus, cond operatorv1.OperatorCondition) (bool, error) {
```

Comment:
Given the usages I think we'll have elsewhere, I think we want the 
```
func UpdateConditionFn(cond operatorv1.OperatorCondition){
	return func(status *operatorv1.StaticPodOperatorStatus) error {
		v1helpers.SetOperatorCondition(&status.Conditions, cond)
		return nil
	}
}
```
so that we can chain the calls and set multiple conditions or the nodestate.

I'm fine with this is you'd like it too

Reply:
yep, also fine. More composable.

Reply:
done